src/App.js

```diff
@@ -14,11 +14,95 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  clearAll = () => {
+    this.setState({ selectedGoods: [] });
+  }
+
+  btnRemove = (good) => {
+    this.setState(prevState => ({
+      selectedGoods: prevState.selectedGoods.filter(
+        product => product !== good,
+      ),
+    }));
+  }
+
+  render() {
+    const { selectedGoods } = this.state;
+
+    return (
+      <div className="App">
+        <h1 className="title">
+          {(() => {
+            switch (selectedGoods.length) {
+              case 0: {
+                return 'No goods selected';
+              }
+
+              case 1: {
+                return `${selectedGoods.join(', ')} is selected`;
+              }
+
+              default: {
+                return `${selectedGoods
+                  .slice(0, (selectedGoods.length - 1))
+                  .join(', ')}`
+                + ` and ${selectedGoods[selectedGoods.length - 1]}`
+                + ` are selected`;
+              }
+            }
+          })()}
+
+          {selectedGoods.length > 0 && (
+            <button
+              className="clear"
+              type="button"
+              onClick={this.clearAll}
+            >
+              乂
+            </button>
+          )}
+        </h1>
+        <div className="wrapper">
+          {goodsFromServer.map(good => (
+            <div className="goods__item" key={good}>
+              <span>
+                {good}
+              </span>
+
+              <button
+                type="button"
+                className={
+                  selectedGoods.includes(good) ? 'good__hidden' : 'btn'
```

Comment:
you can create some variable (`isIncluded` for example) inside callback in `map` method before you return `<li>`, and use this variable in all places where you need it. with this approach, you can use the method includes only once

Reply:
use conditional rendering to hide something